tasktiger/worker.py

```diff
@@ -222,6 +222,14 @@ def _wait_for_new_tasks(self, timeout=0, batch_timeout=0):
            3. Timeout seconds have passed, this is the maximum time to stay in
               this method
         """
+        if not self._pubsub:
+            if self._did_work:
+                self._refresh_queue_set()
+            else:
+                if not self._queue_set:
```

Comment:
Why are we `not self._queue_set`? Is the theory that if we already have some queues we know about we should drain the set before getting new ones? If so we should have a comment here to clarify the logic.

Reply:
I think I can remove the check since the condition `not self._did_work and self._queue_set` should not happen.

In `_worker_run` we call `_process_from_queue` for each queue, which takes a task (or multiple with batch queues) and processes it. It removes the queue from `_queue_set` if it's empty.

In order for `_did_work` to be false, we must have not processed any tasks (or scheduled any tasks). In this case we would have looped through all the queues in the `_queue_set` and removed them.